#05. 트리알고리즘 

로지스틱 회귀로 와인 분류하기 

In [1]:
#데이터 셋 불러오기 
import pandas as pd 
wine = pd.read_csv('https://bit.ly/wine_csv_data')

In [4]:
#와인 데이터셋을 데이터프레임으로 제대로 읽어 들였는지 head()메서드로 5개의 샘플 확인 
wine.head()

alcohol  sugar    pH  class
0      9.4    1.9  3.51    0.0
1      9.8    2.6  3.20    0.0
2      9.8    2.3  3.26    0.0
3      9.8    1.9  3.16    0.0
4      9.4    1.9  3.51    0.0

데이터 셋 설명 

alcohol = 알코올 도수 

sugar = 당도 

pH = pH값 

class = 타깃 값으로 0이면 레드와인, 1이면 화이트 와인 

In [5]:
#데이터프레임의 각 열의 데이터 타입과 누락된 데이터가 있는지 확인
wine.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6497 entries, 0 to 6496
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   alcohol  6497 non-null   float64
 1   sugar    6497 non-null   float64
 2   pH       6497 non-null   float64
 3   class    6497 non-null   float64
dtypes: float64(4)
memory usage: 203.2 KB



=> 총 6,497개의 샘플이 있고 4개의 열은 모두 실숫값 
Non-Null Count가 모두 6497이므로 누락된 값 없음

In [6]:
#열에 대한 간략한 통계를 출력 (최소,최대,평균값)

wine.describe()

alcohol        sugar           pH        class
count  6497.000000  6497.000000  6497.000000  6497.000000
mean     10.491801     5.443235     3.218501     0.753886
std       1.192712     4.757804     0.160787     0.430779
min       8.000000     0.600000     2.720000     0.000000
25%       9.500000     1.800000     3.110000     1.000000
50%      10.300000     3.000000     3.210000     1.000000
75%      11.300000     8.100000     3.320000     1.000000
max      14.900000    65.800000     4.010000     1.000000

mean- 평균 

std - 표준편차 

min - 최소 

25% - 1사분위수 

50% - 중간값/ 2사분위수 

75% - 3사분위수 

max - 최대

In [7]:
# 알콜 도수, 당도, pH 값의 스케일이 다르기에 특성 표준화 

#먼저 넘파이 배열로 바꾸기 
data = wine[['alcohol', 'sugar', 'pH']].to_numpy()
target= wine['class'].to_numpy()

In [8]:
data

array([[ 9.4 ,  1.9 ,  3.51],
       [ 9.8 ,  2.6 ,  3.2 ],
       [ 9.8 ,  2.3 ,  3.26],
       ...,
       [ 9.4 ,  1.2 ,  2.99],
       [12.8 ,  1.1 ,  3.34],
       [11.8 ,  0.8 ,  3.26]])

In [9]:
target

array([0., 0., 0., ..., 1., 1., 1.])

In [10]:
#훈련세트와 테스트 세트로 나눠주기 

from sklearn.model_selection import train_test_split 
train_input, test_input, train_target, test_target = train_test_split(
    data, target, test_size=0.2, random_state =42
)

train_test_splt()함수는 설정값을 지정하지 않으면 25%를 테스트 세트로 지정

샘플개수가 충분히 많으므로 20%정도만 테스트 세트로 나눴음=> test_size =0.2 

In [11]:
print(train_input.shape, test_input.shape)

(5197, 3) (1300, 3)


훈련세트는 5197개, 테스트 세트는 1300개 

In [12]:
#StandardScaler 클래스를 사용해 훈련세트를 전처리 

from sklearn.preprocessing import StandardScaler
ss= StandardScaler()
ss.fit(train_input)
train_scaled= ss.transform(train_input)
test_scaled=ss.transform(test_input)

In [ ]:
from pandas.core.indexers import length_of_indexer
#표준점수로 변환된 train_scaled와 test_scaled를 사용해 로지스틱 회귀모델을 훈련 

from sklearn.linear_model import LogisticRegression
lr = LogisticRegression()
lr.fit(train_scaled, train_target)
print